test/core/crypto/bip39/bip39_integration_test.cpp

```diff
@@ -0,0 +1,261 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <boost/algorithm/string/join.hpp>
+#include "crypto/bip39/impl/bip39_provider_impl.hpp"
+#include "crypto/bip39/mnemonic.hpp"
+#include "crypto/pbkdf2/impl/pbkdf2_provider_impl.hpp"
+#include "testutil/outcome.hpp"
+
+#include <gtest/gtest.h>
+
+using namespace kagome::crypto;
+using namespace bip39;
+
+struct TestItem {
+  std::string mnemonic;
+  std::string entropy;
+  std::string seed;
+};
+
+struct Bip39IntegrationTest : public ::testing::TestWithParam<TestItem> {
+  void SetUp() override {
+    auto pbkdf2_provider = std::make_shared<Pbkdf2ProviderImpl>();
+    bip39_provider = std::make_shared<Bip39ProviderImpl>(pbkdf2_provider);
+  }
+
+  std::shared_ptr<Bip39Provider> bip39_provider;
+};
+
+TEST_P(Bip39IntegrationTest, DeriveEntropyAndSeedSuccess) {
+  const TestItem &item = GetParam();
+  EXPECT_OUTCOME_TRUE(mnemonic, Mnemonic::parse(item.mnemonic));
+  auto joined_words = boost::algorithm::join(mnemonic.words, " ");
+  ASSERT_EQ(joined_words, item.mnemonic);
+
+  EXPECT_OUTCOME_TRUE(entropy,
+                      bip39_provider->calculateEntropy(mnemonic.words));
+
+  std::cout << kagome::common::Buffer(entropy).toHex() << std::endl;
+
+  EXPECT_OUTCOME_TRUE(seed, bip39_provider->makeSeed(entropy, "Substrate"));
+  ASSERT_EQ(seed.toHex(), item.seed);
+}
+
+INSTANTIATE_TEST_CASE_P(
+    SubstrateTestCases,
+    Bip39IntegrationTest,
+    ::testing::Values(
+        TestItem{
+            "abandon abandon abandon abandon abandon abandon abandon abandon "
+            "abandon "
+            "abandon abandon about",
```

Comment:
Pls, remove redundant word wraps for improve human readability (here and other same cases)